cfn-resources/stream-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,230 @@
+	currentModel.Connections = NewModelConnections(streamInstance.Connections)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(UpdateRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	updateRequest := &admin.StreamsDataProcessRegion{
+		CloudProvider: *currentModel.DataProcessRegion.CloudProvider,
+		Region:        *currentModel.DataProcessRegion.Region,
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	_, resp, err := atlasV2.StreamsApi.UpdateStreamInstance(context.Background(), *currentModel.ProjectId, *currentModel.InstanceName, updateRequest).Execute()
+	if err != nil {
+		return handleError(resp, constants.UPDATE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Update success",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(DeleteRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	_, resp, err := atlasV2.StreamsApi.DeleteStreamInstance(context.Background(), *currentModel.ProjectId, *currentModel.InstanceName).Execute()
+	if err != nil {
+		return handleError(resp, constants.DELETE, err)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Delete success"}, nil
+}
+
+func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if errEvent := validator.ValidateModel(ListRequiredFields, currentModel); errEvent != nil {
+		return *errEvent, nil
+	}
+
+	client, handlerError := util.NewAtlasV2OnlyClientLatest(&req, currentModel.Profile, true)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
+		return *handlerError, errors.New(handlerError.Message)
+	}
+
+	atlasV2 := client.AtlasSDK
+
+	accumulatedStreamInstances := make([]admin.StreamsTenant, 0)
+	pageNum := 1
+	for ok := true; ok; {
```

Comment:
wondering if we can use a more specific name (ok usually reminds me of something that didn' return an error instead here it refers to keep calling the API)

Reply:
or perhaps:

```
for {
   ...
   if <END> {
      break;
   }
```

Reply:
I saw this same code is present in @maastha's [PR](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/932/files#) so I am wondering if this is the structure we've always used in other parts of the code...

Reply:
Changed the variable name to `allStreamInstancesRetrieved` and reversed the conditions to match the variable naming. Let me know what you think

Reply:
As per the structure of the code, as far as I know it's the first time we are doing it like this. In some terraform resource(don't remember which one exactly) the retrieval of all elements was being done using recursion, which in my opinion is not the right approach.